src/sentry/static/sentry/app/components/tooltip2.jsx

```diff
@@ -0,0 +1,234 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import styled from 'react-emotion';
+import PropTypes from 'prop-types';
+
+import {Manager, Reference, Popper} from 'react-popper';
+import {domId} from 'app/utils/domId';
+
+class Tooltip2 extends React.Component {
+  static propTypes = {
+    /**
+     * Disable the tooltip display entirely
+     */
+    disabled: PropTypes.bool,
+    /**
+     * The content to show in the tooltip popover
+     */
+    title: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+    /**
+     * Position for the tooltip.
+     */
+    position: PropTypes.oneOf([
+      'bottom',
+      'top',
+      'left',
+      'right',
+      'bottom-start',
+      'bottom-end',
+      'top-start',
+      'top-end',
+      'left-start',
+      'left-end',
+      'right-start',
+      'right-end',
+      'auto',
+    ]),
+    /**
+     * Additional style rules for the tooltip content.
+     */
+    popperStyle: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+  };
+
+  static defaultProps = {
+    position: 'top',
+  };
+
+  constructor(props) {
+    super(props);
+
+    let portal = document.getElementById('tooltip-portal');
+    if (!portal) {
+      portal = document.createElement('div');
+      portal.setAttribute('id', 'tooltip-portal');
+      document.body.appendChild(portal);
+    }
+    this.portalEl = portal;
+  }
+
+  state = {
+    isOpen: false,
+  };
+
+  componentDidMount() {
+    this.tooltipId = domId('tooltip-');
+  }
+
+  handleOpen = evt => {
+    this.setState({isOpen: true});
+  };
+
+  handleClose = evt => {
+    this.setState({isOpen: false});
+  };
+
+  /**
+   * Different kinds of children need different wrapping.
+   * For custom components and function components we have
+   * to add a span as we can't trust the component to implement
+   * forward refs. For svg and html elements we can clone
+   * the child and push more props onto it.
+   */
+  renderTrigger(children, ref) {
+    const propList = {
+      'aria-describedby': this.tooltipId,
+      onFocus: this.handleOpen,
+      onBlur: this.handleClose,
+      onMouseEnter: this.handleOpen,
+      onMouseLeave: this.handleClose,
+    };
+    if (children.type instanceof Function) {
```

Comment:
Let's re-state our assumptions about what we're expecting for `children.type` here.